airflow/lineage/__init__.py

```diff
@@ -110,26 +111,32 @@ def wrapper(self, context, *args, **kwargs):
                       for i in inlets]
             self.inlets.extend(inlets)
 
-        if self._inlets['auto']:
-            # dont append twice
-            task_ids = set(self._inlets['task_ids']).symmetric_difference(
-                self.upstream_task_ids
-            )
-            inlets = self.xcom_pull(context,
-                                    task_ids=task_ids,
-                                    dag_id=self.dag_id,
-                                    key=PIPELINE_OUTLETS)
-            inlets = [item for sublist in inlets if sublist for item in sublist]
-            inlets = [DataSet.map_type(i['typeName'])(data=i['attributes'])
-                      for i in inlets]
-            self.inlets.extend(inlets)
-
-        if len(self._inlets['datasets']) > 0:
-            self.inlets.extend(self._inlets['datasets'])
+        if self._inlets["auto"]:  # Performs a tree traversal, starting with the current task. If outlets
```

Comment:
Looks like, it should be multiline docsting https://www.python.org/dev/peps/pep-0257/#multi-line-docstrings

Reply:
Sure! I wanted to keep the documentation as is since I was writing this PR to match the features of the documentation. On hindsight, since since it does add behaviour not fully covered by the existing docstring, it makes sense to move it into the docstring and clear the behaviour up a little.